### PR TITLE
Update for use with python 3.6

### PR DESCRIPTION
**Pull request by ryanmrubin**
Altenative to #6 .

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 behave==1.2.5
-Markdown==2.5.2  # FIXME: 2.6.8 breaks tests
+Markdown==2.6.8
```
Comment (eykd) [inline requirements-dev.txt]: I forget why this change was breaking tests earlier, but you may want to address everything else before upgrading this.

Comment (ryanmrubin) [inline requirements-dev.txt]: I did try it without that change, an there's one extra failing test under this--it's not the widespread problem we saw before, so I think I'll leave it for now.

File: features/steps/markdown2html_steps.py [hunk 2]
```diff
@@ -15,11 +13,8 @@
 def normalize_html(html_text):
     html_text = '\n'.join(html.tostring(el, encoding=str)
                            for el in antimarkdown.parse_fragments(html_text))
-    tidied = tidy(
-        '\n'.join(line.strip()
-                   for line in nodes.whitespace(html_text.strip()).replace('>', '>\n').splitlines()),
-        pretty_print=True)
-    return tidied.decode('utf-8')
+    return '\n'.join(line.strip()
+                     for line in nodes.whitespace(html_text.strip()).replace('>', '>\n').splitlines())
```
Comment (eykd) [inline features/steps/markdown2html_steps.py]: You're probably going to want some sort of `tidy()` function so you get sane diffs for any broken tests. BeautifulSoup has decent tree normalization, or you could use Derek's [tidy-alike](https://github.com/Crossway/antimarkdown/pull/6/files#diff-463b9c97c640c5e992540b7ca49b3902R17), though I haven't evaluated its effectiveness yet.

File: antimarkdown/nodes.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 hex_color = collections.defaultdict(lambda: 'none')
 try:
     import mdx_hilite
-    hex_color.update((h, c) for c, h in mdx_hilite.color_hex.iteritems())
+    hex_color.update((h, c) for c, h in mdx_hilite.color_hex.items())
```
Comment (eykd) [inline antimarkdown/nodes.py]: We're dropping support for `mdx_hilite` so this whole business can probably be removed altogether.

File: features/steps/data/__init__.py [hunk 4]
```diff
@@ -60,7 +60,7 @@ def get_args(file, config):
 
 def normalize(text):
     """ Normalize whitespace for a string of html using tidy. """
-    return str(tidy.parseString(text.encode('utf-8', 'xmlcharrefreplace'), 
+    return str(tidy.parseString(text.encode('utf-8', 'xmlcharrefreplace'),
```
Comment (eykd) [inline features/steps/data/__init__.py]: We no longer have `tidy` to work with.

File: features/steps/data/__init__.py [hunk 5]
```diff
@@ -10,10 +10,8 @@
                       "to install the latest version.")
 from . import util
 from .plugins import HtmlOutput, Markdown
-try:
-    import tidy
-except ImportError:
-    tidy = None
+
+tidy = None
```
Comment (eykd) [inline features/steps/data/__init__.py]: I think we need to address the lack of `tidy` wholesale, rather than relying on this being `None`.

Comment (ryanmrubin) [inline features/steps/data/__init__.py]: We will. I just wanted to get this up with a minimal incision so as to discuss with you.

File: features/steps/data/__init__.py (hunk 4 above)
Comment (ryanmrubin) [inline features/steps/data/__init__.py]: ah ha. missed one.

File: features/steps/markdown2html_steps.py (hunk 2 above)
Comment (eykd) [inline features/steps/markdown2html_steps.py]: Ah, maybe just use bs4's [pretty-printing](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#pretty-printing).

File: antimarkdown/nodes.py (hunk 3 above)
Comment (ryanmrubin) [inline antimarkdown/nodes.py]: ✅

File: README.rst [hunk 6]
```diff
@@ -6,11 +6,11 @@ Convert HTML to Markdown, quickly and easily:
 ::
 
     >>> import antimarkdown
-    >>> print antimarkdown.to_markdown("""
+    >>> print(antimarkdown.to_markdown("""
     ... <h1>antimarkdown</h1>
     ...
     ... <p>Convert HTML to Markdown, quickly and easily!</p>
-    ... """)
+    ... """))_
```
Comment (eykd) [inline README.rst]: Maybe no trailing underscore? 😁

File: antimarkdown/nodes.py [hunk 7]
```diff
@@ -208,115 +202,106 @@ class LI(Block):
...
-            li = li.next()
+        li = self.blackboard.get('li-style', ['*   '])[-1]
+        if inspect.isgenerator(li):
+            li = next(li)
         text = whitespace(eltext(self.el.text)).lstrip()
 
-        lines = newlines(u''.join(u'\n' + unicode(node)
-                                  if isinstance(node, Block) else unicode(node)
+        lines = newlines(''.join('\n' + str(node)
+                                  if isinstance(node, Block) else str(node)
```
Comment (eykd) [inline antimarkdown/nodes.py]: The indentation here and in the next line are one space too many after dropping the `u` wart from the joining string.

File: setup.py [hunk 8]
```diff
@@ -12,15 +12,14 @@
...
-        'BeautifulSoup',
     ],
 
     package_data = {
         '': ['*.txt', '*.html'],
     },
     zip_safe = False,
 
-    version = "0.2.1",
+    version = "1.0.0",
```
Comment (eykd) [inline setup.py]: Oh heck, why not?

Comment (ryanmrubin) [inline setup.py]: hehe

File: README.rst (hunk 6 above)
Comment (ryanmrubin) [inline README.rst]: Whoops!
